kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -182,6 +182,7 @@ private[kyuubi] class EngineRef(
           }
         }
         if (started + timeout <= System.currentTimeMillis()) {
+          builder.closeYarnJob()
```

Comment:
Maybe `killApplication` is a more suitable as engine may be submitted to other Cluster Managers such as kubernetes.